errors/error.go

```diff
@@ -0,0 +1,194 @@
+	DivideByZero = ErrorKind("ErrDivideByZero")
+
+	// HexLength indicates an invalid hex length.
+	HexLength = ErrorKind("ErrHexLength")
+
+	// TxConf indicates a transaction confirmation error.
+	TxConf = ErrorKind("ErrTxConf")
+
+	// BlockConf indicates a block confirmation error.
+	BlockConf = ErrorKind("ErrBlockConf")
+
+	// ClaimShare indicates a share claim error.
+	ClaimShare = ErrorKind("ErrClaimShare")
+
+	// LimitExceeded indicates a rate limit exhaustion error.
+	LimitExceeded = ErrorKind("ErrLimitExceeded")
+
+	// Difficulty indicates a difficulty related error.
+	Difficulty = ErrorKind("ErrDifficulty")
+
+	// WorkRejected indicates the rejected submitted work.
+	WorkRejected = ErrorKind("ErrWorkRejected")
+
+	// PaymentSource indicates a payment source error.
+	PaymentSource = ErrorKind("ErrPaymentSource")
+
+	// ShareRatio indicates a share ratio error.
+	ShareRatio = ErrorKind("ErrShareRatio")
+
+	// CreateHash indicates a hash creation error.
+	CreateHash = ErrorKind("ErrCreateHash")
+
+	// Coinbase indicates a coinbase related error.
+	Coinbase = ErrorKind("ErrCoinbase")
+
+	// CreateTx indicates a transaction creation error.
+	CreateTx = ErrorKind("ErrCreateTx")
+
+	// SignTx indicates a transaction signing error.
+	SignTx = ErrorKind("ErrSignTx")
+
+	// PublishTx indicates a transaction pubishing error.
+	PublishTx = ErrorKind("ErrPublishTx")
+
+	// TxOut indicates a transaction output related error.
+	TxOut = ErrorKind("ErrTxOut")
+
+	// TxIn indicates a transaction input related error.
+	TxIn = ErrorKind("ErrTxIn")
+
+	// ContextCancelled indicates a context cancellation related error.
+	ContextCancelled = ErrorKind("ErrContextCancelled")
+
+	// CreateAmount indicates an amount creation error.
+	CreateAmount = ErrorKind("ErrCreateAmount")
+)
+
+// Error satisfies the error interface and prints human-readable errors.
+func (e ErrorKind) Error() string {
+	return string(e)
+}
+
+// Error identifies an error. It has full support for errors.Is and
+// errors.As, so the caller can ascertain the specific reason for
+// the error by checking the underlying error.
+type Error struct {
+	Description string
+	Err         error
+}
+
+// New creates a simple error from a string.  New is identical to "errors".New
+// from the standard library.
+func New(text string) error {
+	return errors.New(text)
+}
+
+// Error satisfies the error interface and prints human-readable errors.
+func (e Error) Error() string {
+	return e.Description
+}
+
+// Unwrap returns the underlying wrapped error.
+func (e Error) Unwrap() error {
+	return e.Err
+}
+
+// Is returns whether err equals or wraps target.
+func Is(err, target error) bool {
+	return errors.Is(err, target)
+}
+
+// As attempts to assign the error pointed to by target with the first error in
+// err's error chain with a compatible type.  Returns true if target is
+// assigned.
+func As(err error, target interface{}) bool {
+	return errors.As(err, target)
+}
```

Comment:
Hmm, looking at the call sites for these i think a namespace alias is a better fit than these funcs.